src/libstd/thread/mod.rs

```diff
@@ -909,9 +909,10 @@ pub fn park() {
     }
     loop {
         m = thread.inner.cvar.wait(m).unwrap();
-        match thread.inner.state.compare_exchange(NOTIFIED, EMPTY, SeqCst, SeqCst) {
-            Ok(_) => return, // got a notification
-            Err(_) => {} // spurious wakeup, go back to sleep
+
+        // Return notification or do nothing on spurious notifications
+        if let Ok(_) = thread.inner.state.compare_exchange(NOTIFIED, EMPTY, SeqCst, SeqCst) {
+            return;
```

Comment:
I'm not entirely sure about this one. Alternatively I can just make Clippy ignore it.

Reply:
I am confused -- what are you unsure about?

Reply:
I'm not sure if new version is more readable, should have expressed it more clearly.

Reply:
How about keeping the `if`, but still having the two comments?  Maybe keep "got a notification" before the return, and move "spurious wakeup, go back to sleep" to the end of the block?

Reply:
IMO it wasn't any better, dropped this change.